index.js

```diff
@@ -77,12 +77,13 @@ var exec = function (cmd, opts, cb) {
         stream.emit('warn', data)
       })
 
-      stdio.on('close', function () {
-        client.end()
-      })
-
       stdio.on('exit', function (code) {
-        stream.emit('exit', code)
+        if (code !== 0) {
+          stream.emit('error', code)
```

Comment:
this should emit an actual error, `var err = new Error('Non-zero exit code: ' + code)` and `err.code = code`.